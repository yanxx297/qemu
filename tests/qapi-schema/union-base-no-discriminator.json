# FIXME: we should reject simple unions with a base
{ 'type': 'TestTypeA',
  'data': { 'string': 'str' } }

{ 'type': 'TestTypeB',
  'data': { 'integer': 'int' } }

{ 'type': 'Base',
  'data': { 'string': 'str' } }

{ 'union': 'TestUnion',
  'base': 'Base',
  'data': { 'value1': 'TestTypeA',
            'value2': 'TestTypeB' } }
