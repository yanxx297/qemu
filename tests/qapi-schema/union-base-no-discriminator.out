[OrderedDict([('type', 'TestTypeA'), ('data', OrderedDict([('string', 'str')]))]),
 OrderedDict([('type', 'TestTypeB'), ('data', OrderedDict([('integer', 'int')]))]),
 OrderedDict([('type', 'Base'), ('data', OrderedDict([('string', 'str')]))]),
 OrderedDict([('union', 'TestUnion'), ('base', 'Base'), ('data', OrderedDict([('value1', 'TestTypeA'), ('value2', 'TestTypeB')]))])]
[{'enum_name': 'TestUnionKind', 'enum_values': None}]
[OrderedDict([('type', 'TestTypeA'), ('data', OrderedDict([('string', 'str')]))]),
 OrderedDict([('type', 'TestTypeB'), ('data', OrderedDict([('integer', 'int')]))]),
 OrderedDict([('type', 'Base'), ('data', OrderedDict([('string', 'str')]))])]
